### PR TITLE
build: remove buster, replace with bullseye when necessary

### PR DESCRIPTION
**Pull request by aminvakil**
It's been three years since buster does not gets updated from Debian Security Team and it's been a year since Debian LTS Team does not support that either.

Now that `trixie` has been added as well to this repository, I've removed all `buster` tests.

There were also some tests which only were tested on `buster`, I've changed them to `bullseye` for now to expect minimum changes, and I'm going to change them to `trixie` with is Debian `stable` now.

----
Enter `` in the box, if an item is not applicable to your change.

**Testing**
Before we can approve your change; please submit the following in a comment:

- [N/A] Example configuration file for the change
- [N/A] Debug log output from testing the change
- [N/A] Attached [Valgrind](https://valgrind.org/docs/manual/quick-start.html) output that shows no leaks or memory corruption was found

If this is a change to packaging of containers or native binaries then please confirm it works for all targets.

- [ ] Run [local packaging test](./packaging/local-build-all.sh) showing all targets (including any new ones) build.
- [ ] Set `ok-package-test` label to test for all targets (requires maintainer to do).

**Documentation**

- [x] Documentation required for this feature

https://github.com/fluent/fluent-bit-docs/pull/2068

**Backporting**

- [N/A] Backport to latest stable release.

----

Fluent Bit is licensed under Apache 2.0, by submitting this pull request I understand that this code will be released under the terms of that license.

## Summary by CodeRabbit

* **Chores**
  * Switched build pipelines from Debian Buster to Debian Bullseye across releases.
  * Removed Debian Buster build targets and repository entries.

* **Documentation**
  * Updated packaging documentation and comments to reflect removal of Debian Buster (and Raspbian Buster) targets.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by patrick-stephens — COMMENTED** [review]
I would disable the build but not remove the option for folks to build, I know it's eol but that's not the same as no one uses it (look at Centos). We can just remove it from all the CI but leave the container definition I think.

**Comment by aminvakil** [comment]
> I would disable the build but not remove the option for folks to build, I know it's eol but that's not the same as no one uses it (look at Centos). We can just remove it from all the CI but leave the container definition I think.

https://github.com/fluent/fluent-bit/pull/10908/commits/65c15b4a846afb5ef308e60a40327ded1552a50a

Does this suffice?

File: .github/actions/generate-package-build-matrix/action.yaml [hunk 1]
```diff
@@ -67,18 +67,18 @@ runs:
           echo '{ "distro" : ['
           echo '"amazonlinux/2", "amazonlinux/2.arm64v8",'
           echo '"centos/7", "centos/7.arm64v8", "centos/8", "centos/8.arm64v8",'
-          echo '"debian/buster", "debian/buster.arm64v8", "debian/bullseye", "debian/bullseye.arm64v8",'
```
Comment (patrick-stephens) [inline .github/actions/generate-package-build-matrix/action.yaml:70]: No need to change this, it's only for 1.9 and earlier builds

File: packaging/distros/raspbian/Dockerfile [hunk 2]
```diff
@@ -6,7 +6,7 @@ ARG BASE_BUILDER
 # Lookup the name to use below but should follow the '<distro>-base' convention with slashes replaced.
 # Use buildkit to skip unused base images: DOCKER_BUILDKIT=1
 
-FROM balenalib/rpi-raspbian:buster AS raspbian-buster-base
+FROM balenalib/rpi-raspbian:bullseye AS raspbian-bullseye-base
```
Comment (patrick-stephens) [inline packaging/distros/raspbian/Dockerfile]: This is actually changing the Raspbian build, I'm not sure we want that as it's not Debian per-se. Not sure why we jumped from buster to bookworm for Raspbian but that's not part of this change.

Please make sure you're not just globally changing any reference to the string.

File: packaging/testing/smoke/packages/Dockerfile.debian10 [hunk 3]
```diff
@@ -1,11 +1,11 @@
 # For staging upgrade we use the 'official-install' as the base
-ARG STAGING_BASE=docker.io/dokken/debian-10
+ARG STAGING_BASE=docker.io/dokken/debian-11
```
Comment (patrick-stephens) [inline packaging/testing/smoke/packages/Dockerfile.debian10]: Again, this is a change that is confusing as the file we are in here is called `.debian10`, I think it's been a global search-and-replace when instead we should also rename the file to be clear and accurate.

File: .github/actions/generate-package-build-matrix/action.yaml (hunk 1 above)
Comment (aminvakil) [reply .github/actions/generate-package-build-matrix/action.yaml:70]: Reverted

File: packaging/distros/raspbian/Dockerfile (hunk 2 above)
Comment (aminvakil) [reply packaging/distros/raspbian/Dockerfile]: Reverted.

File: packaging/testing/smoke/packages/Dockerfile.debian10 (hunk 3 above)
Comment (aminvakil) [reply packaging/testing/smoke/packages/Dockerfile.debian10]: Sorry, this was a mistake and I have reverted this.

Yes. It was a global search-and-replace. I wasn't sure if this file is still needed, so I have not removed it and just reverted my wrong change.

**Comment by aminvakil** [comment]
None of failing tests are related to this PR.

**Comment by eschabell** [comment]
@patrick-stephens can you approve this to get it merged? 

Docs PR https://github.com/fluent/fluent-bit-docs/pull/2068 ready to go.

File: src/wamrc/CMakeLists.txt [hunk 4]
```diff
@@ -252,4 +252,4 @@ if (LLVM_FOUND)
     install(FILES $<TARGET_PDB_FILE:flb-wamrc-bin>
       DESTINATION "${FLB_INSTALL_BINDIR}")
   endif()
-endif ()
\ No newline at end of file
+endif ()
```
Comment (patrick-stephens) [inline src/wamrc/CMakeLists.txt]: You should not be changing this, it's a vendored dependency so unrelated to this change.

Comment (aminvakil) [reply src/wamrc/CMakeLists.txt]: I'm pretty sure this is vim vs vscode default behvaiour with newline, I didn't intend to change this :)

I'll revert this.

**Comment by patrick-stephens** [comment]
@aminvakil can you update your commits to match the required contribution style (or squash them)

File: lib/cfl/.github/workflows/build.yaml [hunk 5]
```diff
@@ -100,9 +100,9 @@ jobs:
         working-directory: cfl
 
   build-debian:
-    name: Debian Buster build to confirm no issues once used downstream
+    name: Debian Bullseye build to confirm no issues once used downstream
     runs-on: ubuntu-latest
-    container: debian:buster
+    container: debian:bullseye
```
Comment (cosmo0920) [inline lib/cfl/.github/workflows/build.yaml]: You shouldn't modify under lib sources. This should be modified in each of upstream repos.

File: lib/cmetrics/.github/workflows/build.yaml [hunk 6]
```diff
@@ -84,9 +84,9 @@ jobs:
         working-directory: cmetrics
 
   build-debian:
-    name: Debian Buster build to confirm no issues once used downstream
+    name: Debian Bullseye build to confirm no issues once used downstream
     runs-on: ubuntu-latest
-    container: debian:buster
+    container: debian:bullseye
```
Comment (cosmo0920) [inline lib/cmetrics/.github/workflows/build.yaml]: Ditto.

File: lib/cprofiles/.github/workflows/build.yaml [hunk 7]
```diff
@@ -75,9 +75,9 @@ jobs:
         working-directory: cprofiles/build
 
   build-debian:
-    name: Debian Buster build to confirm no issues once used downstream
+    name: Debian Bullseye build to confirm no issues once used downstream
     runs-on: ubuntu-latest
-    container: debian:buster
+    container: debian:bullseye
```
Comment (cosmo0920) [inline lib/cprofiles/.github/workflows/build.yaml]: Ditto.

File: lib/ctraces/.github/workflows/build.yaml [hunk 8]
```diff
@@ -79,9 +79,9 @@ jobs:
         working-directory: ctraces
 
   build-debian:
-    name: Debian Buster build to confirm no issues once used downstream
+    name: Debian Bullseye build to confirm no issues once used downstream
     runs-on: ubuntu-latest
-    container: debian:buster
+    container: debian:bullseye
```
Comment (cosmo0920) [inline lib/ctraces/.github/workflows/build.yaml]: Ditto.

**Review by cosmo0920 — CHANGES_REQUESTED** [review]
We'll modify cfl, cmetrics, cprofiles, and ctraces in each of repositories.
So, in this PR, we shouldn't modify them directly.

File: lib/cfl/.github/workflows/build.yaml (hunk 5 above)
Comment (aminvakil) [reply lib/cfl/.github/workflows/build.yaml]: Reverted in github.com/fluent/fluent-bit/pull/10908/commits/392073578dcf5e5e862ac2926e62c60b92b28c04.

File: lib/cmetrics/.github/workflows/build.yaml (hunk 6 above)
Comment (aminvakil) [reply lib/cmetrics/.github/workflows/build.yaml]: Reverted in github.com/fluent/fluent-bit/pull/10908/commits/392073578dcf5e5e862ac2926e62c60b92b28c04.

File: lib/cprofiles/.github/workflows/build.yaml (hunk 7 above)
Comment (aminvakil) [reply lib/cprofiles/.github/workflows/build.yaml]: Reverted in github.com/fluent/fluent-bit/pull/10908/commits/392073578dcf5e5e862ac2926e62c60b92b28c04.

File: lib/ctraces/.github/workflows/build.yaml (hunk 8 above)
Comment (aminvakil) [reply lib/ctraces/.github/workflows/build.yaml]: Reverted in github.com/fluent/fluent-bit/pull/10908/commits/392073578dcf5e5e862ac2926e62c60b92b28c04.

**Review by patrick-stephens — APPROVED** [review]
